<?xml version="1.0"?>

<launch>

  <arg name="config_file" default="$(dirname)/../config/mapping_default.yaml"/>

  <arg name="map_frame"  default="map"/>
  <arg name="base_frame" default="base_footprint"/>
  <arg name="odom_frame" default="nav"/>

  <arg name="use_tf_scan_transformation" default="true"/>
  <arg name="use_tf_pose_start_estimate" default="false"/>
  <arg name="pub_map_odom_transform"     default="true"/>

  <arg name="map_resolution"       default="0.050"/>
  <arg name="map_size"             default="2048"/>
  <arg name="map_start_x"          default="0.5"/>
  <arg name="map_start_y"          default="0.5"/>
  <arg name="map_multi_res_levels" default="2"/>

  <arg name="scan_subscriber_queue_size" default="5"/>
  <arg name="scan_topic"                 default="scan"/>

  <arg name="tf_map_scanmatch_transform_frame_name" default="scanmatcher_frame"/>



  <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">

    <rosparam command="load" file="$(arg config_file)" subst_value="true"/>

  </node>

  <!--<node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster" args="0 0 0 0 0 0 map nav 100"/>-->
</launch>
